### PR TITLE
VZV | Update time of day chart

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/3555

This PR updates the Time of Day chart to have the same maximum value across each year for each type of crash (All, Fatalities, Serious Injuries). Since only a single year's data is fetched at once, I added a SoQL query to get hour and days totals across the entire VZ rolling time window for each type and then sub-query for the maximum of each of those hourly, daily totals. This query will be reusable if we decide to refactor the chart data fetch in the future.

The legend is then set with that max and a placeholder column filled with the max value is added to the chart data. That column is then hidden and the left padding is adjusted to center the chart.

Thanks to @mateoclarke for the idea about the extra column with the maximums. Worked like magic!

## Before
Screen Shot 2020-08-26 at 11 17 49 AM
Screen Shot 2020-08-26 at 11 17 53 AM

## After
Screen Shot 2020-08-26 at 11 17 59 AM
Screen Shot 2020-08-26 at 11 18 03 AM

### CODE REVIEW

File: atd-vzv/src/views/summary/CrashesByTimeOfDay.js [hunk 1]
```diff
@@ -96,13 +101,77 @@ const CrashesByTimeOfDay = () => {
...
+    const maxQuery = `
+    SELECT date_extract_dow(crash_date) as day, date_extract_hh(crash_date) as hour, date_extract_y(crash_date) as year, SUM(death_cnt) as death, SUM(sus_serious_injry_cnt) as serious, serious + death as all 
+    WHERE crash_date BETWEEN '${dataStartDate.format(
+      "YYYY-MM-DD"
+    )}' and '${summaryCurrentYearEndDate}' 
+    GROUP BY day, hour, year 
+    ORDER BY year 
+    |> 
+    SELECT max(death) as fatalities, max(serious) as seriousInjuries, max(all) as fatalitiesAndSeriousInjuries
+    `;
```
Comment (sergiogcx) [inline atd-vzv/src/views/summary/CrashesByTimeOfDay.js:123]: Very cool to see that chain operator `|>` learning something cool!

File: atd-vzv/src/views/summary/CrashesByTimeOfDay.js [hunk 2]
```diff
@@ -96,13 +101,77 @@ const CrashesByTimeOfDay = () => {
...
+        `#demographics-heatmap > div > svg > g > g:nth-child(${num})`
+      )
+    );
+
+    cellsToHide.forEach((cell) => {
+      if (!!cell) {
+        cell.style.visibility = "hidden";
+      }
+    });
+  }, [heatmapDataWithPlaceholder, crashType, maxForLegend]);
```
Comment (sergiogcx) [inline atd-vzv/src/views/summary/CrashesByTimeOfDay.js:174]: Hacking the matrix! love that document.querySelector use

File: atd-vzv/src/views/summary/CrashesByTimeOfDay.js (hunk 1 above)
Comment (mddilley) [reply atd-vzv/src/views/summary/CrashesByTimeOfDay.js:123]: Yes! I was happy to find a use for this hidden gem.

Comment (mateoclarke) [reply atd-vzv/src/views/summary/CrashesByTimeOfDay.js:123]: ✨

File: atd-vzv/src/views/summary/CrashesByTimeOfDay.js (hunk 2 above)
Comment (mateoclarke) [reply atd-vzv/src/views/summary/CrashesByTimeOfDay.js:174]: Really glad this idea panned out! Great work Mike!

**Pull request merged**